service/history/api/respondactivitytaskfailed/api_test.go

```diff
@@ -0,0 +1,549 @@
+ * // THE SOFTWARE.
+ */
+
+package respondactivitytaskfailed
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	commonpb "go.temporal.io/api/common/v1"
+	enums2 "go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/workflowservice/v1"
+
+	"go.temporal.io/server/api/enums/v1"
+	"go.temporal.io/server/api/historyservice/v1"
+	"go.temporal.io/server/api/persistence/v1"
+	tokenspb "go.temporal.io/server/api/token/v1"
+	"go.temporal.io/server/common"
+	"go.temporal.io/server/common/clock"
+	"go.temporal.io/server/common/cluster"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/metrics"
+	"go.temporal.io/server/common/namespace"
+	persistence2 "go.temporal.io/server/common/persistence"
+	"go.temporal.io/server/common/primitives/timestamp"
+	"go.temporal.io/server/service/history/api"
+	"go.temporal.io/server/service/history/consts"
+	"go.temporal.io/server/service/history/shard"
+	"go.temporal.io/server/service/history/tests"
+	"go.temporal.io/server/service/history/workflow"
+	wcache "go.temporal.io/server/service/history/workflow/cache"
+)
+
+type (
+	workflowSuite struct {
+		suite.Suite
+		*require.Assertions
+
+		controller        *gomock.Controller
+		shardContext      *shard.MockContext
+		namespaceRegistry *namespace.MockRegistry
+
+		workflowCache              *wcache.MockCache
+		workflowConsistencyChecker api.WorkflowConsistencyChecker
+
+		currentContext      *workflow.MockContext
+		currentMutableState *workflow.MockMutableState
+
+		activityInfo *persistence.ActivityInfo
+	}
+)
+
+func TestWorkflowSuite(t *testing.T) {
+	s := new(workflowSuite)
+	suite.Run(t, s)
+}
+
+func (s *workflowSuite) SetupSuite() {
+}
+
+func (s *workflowSuite) TearDownSuite() {
+}
+
+func (s *workflowSuite) SetupTest() {
+	s.Assertions = require.New(s.T())
+	s.controller = gomock.NewController(s.T())
+}
+
+func (s *workflowSuite) TearDownTest() {
+	s.controller.Finish()
+}
+
+type UsecaseConfig struct {
+	request             *historyservice.RespondActivityTaskFailedRequest
+	attempt             int32
+	activityId          string
+	activityType        string
+	startedEventId      int64
+	scheduledEventId    int64
+	taskQueueId         string
+	isActivityActive    bool
+	isExecutionRunning  bool
+	expectRetryActivity bool
+	retryActivityError  error
+	retryActivityState  enums2.RetryState
+	namespaceId         namespace.ID
+	namespaceName       namespace.Name
+	wfType              *commonpb.WorkflowType
+	tokenVersion        int64
+	tokenAttempt        int32
+	isCacheStale        bool
+	includeHeartbeat    bool
+}
+
+func (s *workflowSuite) Test_NormalFlowShouldRescheduleActivity_UpdatesWorkflowExecutionAsActive() {
+	ctx := context.Background()
+	uc := newUseCase(UsecaseConfig{
+		attempt:             int32(1),
+		startedEventId:      int64(40),
+		scheduledEventId:    int64(42),
```

Comment:
Are `attempt`, `startedEventId` and `scheduledEventId` always the same across tests? If so, should they be set inside of a helper function instead?